checks/phpcs.php

```diff
@@ -110,21 +110,133 @@ protected function phpcs_result_to_warnings( $result ) {
 				}
 
 				$source_code = esc_html( trim( file( $this->path . '/' . $filename )[ $message['line'] - 1 ] ) );
+				$edit_link   = sprintf(
+					'<a href="%1$s" title="%2$s" aria-label="%2$s" target="_blank">%3$s</a>',
+					$this->get_file_editor_url( $filename, $message['line'] ),
+					sprintf(
+						/* translators: %s is the path to a plugin file. */
+						esc_attr__( 'View %s in the plugin file editor.', 'plugin-check' ),
+						$this->slug . '/' . $filename
+					),
+					esc_html__( 'View in code editor', 'plugin-check' )
+				);
 
 				$return[] = new $notice_class(
 					$message['source'],
 					sprintf(
-						'%s Line %d of file %s.<br>%s.<br>%s',
+						'%s Line %d of file %s.<br>%s.<br>%s%s',
 						"<strong>{$message['source']}</strong>",
 						$message['line'],
 						$filename,
 						rtrim( $message['message'], '.' ),
-						"<pre><code>{$source_code}</code></pre>"
+						"<pre><code>{$source_code}</code></pre>",
+						$edit_link
 					)
 				);
 			}
 		} );
 
 		return $return;
 	}
+
+	/**
+	 * Get the URL for opening the plugin file in an external editor.
+	 *
+	 * @since 0.2.1
+	 *
+	 * @param array $filename Source of PHPCS error.
+	 * @param array $line     Line number of PHPCS error.
+	 *
+	 * @return string|null File editor URL or null if not available.
+	 */
+	private function get_file_editor_url( $filename, $line ) {
+		if ( ! isset( $filename, $line ) ) {
+			return null;
+		}
+
+		$edit_url = null;
+
+		/**
+		 * Filters the template for the URL for linking to an external editor to open a file for editing.
+		 *
+		 * Users of IDEs that support opening files in via web protocols can use this filter to override
+		 * the edit link to result in their editor opening rather than the plugin editor.
+		 *
+		 * The initial filtered value is null, requiring extension plugins to supply the URL template
+		 * string themselves. If no template string is provided, links to the plugin editors will
+		 * be provided if available. For example, for an extension plugin to cause file edit links to
+		 * open in an IDE, the following filters can be used:
+		 *
+		 * # PhpStorm
+		 * add_filter( 'pcp_validation_error_source_file_editor_url_template', function () {
+		 *     return 'phpstorm://open?file={{file}}&line={{line}}';
+		 * } );
+		 *
+		 * # VS Code
+		 * add_filter( 'pcp_validation_error_source_file_editor_url_template', function () {
+		 *     return 'vscode://file/{{file}}:{{line}}';
+		 * } );
+		 *
+		 * For a template to be considered, the string '{{file}}' must be present in the filtered value.
+		 *
+		 * @since 0.2.1
+		 *
+		 * @param string|null $editor_url_template Editor URL template.
+		 */
+		$editor_url_template = apply_filters( 'pcp_validation_error_source_file_editor_url_template', null );
```

Comment:
@bordoni I wasn't sure what the best name to use here was. I chose pcp_ (plugin check plugin) as the prefix, but we can swap to `plugin_check_` or something else if it makes more sense.

Reply:
Done in [1e66025](https://github.com/10up/plugin-check/pull/262/commits/1e66025fc040b8c75cd7a40f8622ff21618efbac)